src/mixins/README.md

```diff
@@ -0,0 +1,82 @@
+# Mixins
+
+This directory contains mixins for `Application`.
+
+## Overview
+
+A mixin is a class which take the input of a base class and adds / modifies existing functions / status values and returns a new class which can be instantiated. The type of the returned class will still be the base class. Mixins let you extend the BaseClass by adding more functions to it, overriding existing ones or modifiying their behavior.
```

Comment:
I am concerned that the content in this README is missing a clear explanation why extension developers would want/need to write an application mixin, and what are the necessary steps/features that such mixin should do in order to be consistent with other similar mixins. I would like to see a short paragraph explaining the different ways how a configuration can be provided to a LoopBack app (application config, component config, `app.logger()` helper) and perhaps explaining how to implement each option via the mixin.

On the second thought, I think such content should be in http://loopback.io/doc/en/lb4/Creating-components.html and this README should link to it.

> The type of the returned class will still be the base class

This is not true, the returned class will be a child class inheriting from the original base class.